ironfish-rust/src/transaction/evm.rs

```diff
@@ -35,26 +27,89 @@ impl EvmDescription {
         to: Option<[u8; 20]>,
         value: u64,
         data: Vec<u8>,
-        v: u8,
-        r: [u8; 32],
-        s: [u8; 32],
         private_iron: u64,
         public_iron: u64,
     ) -> Self {
-        Self {
+        let description = EvmDescription {
             nonce,
             gas_price,
             gas_limit,
             to,
             value,
             data,
-            v,
-            r,
-            s,
+            v: 0,
+            r: [0u8; 32],
+            s: [0u8; 32],
             private_iron,
             public_iron,
-        }
+        };
+
+        Self { description }
     }
+
+    pub fn read<R: io::Read>(reader: R) -> Result<Self, io::Error> {
+        let description = EvmDescription::read(reader)?;
+
+        Ok(Self { description })
+    }
+
+    pub fn write<W: io::Write>(&self, writer: W) -> Result<(), IronfishError> {
+        EvmDescription::write(&self.description, writer)
+    }
+
+    pub(crate) fn serialize_signature_fields<W: io::Write>(
+        &self,
+        mut writer: W,
+    ) -> Result<(), IronfishError> {
+        self.description.serialize_signature_fields(&mut writer)
+    }
+
+    pub fn sign(mut self, spender_key: &SaplingKey) -> EvmDescription {
+        let tx_kind = match self.description.to {
+            None => TxKind::Create,
+            Some(address_bytes) => TxKind::Call(Address::from(address_bytes)),
+        };
+        let reth_tx: Transaction = Transaction::Legacy(TxLegacy {
+            nonce: self.description.nonce,
+            gas_price: self.description.gas_price.into(),
+            gas_limit: self.description.gas_limit,
+            to: tx_kind,
+            value: U256::from(self.description.value),
+            input: self.description.data.clone().into(),
+            chain_id: None,
+        });
+
+        // TODO(hughy): add appropriate error handling for signing errors from reth
```

Comment:
does this need to be tackled here?

Reply:
I think I'm going to leave the unwrap for now 😅